src/main/java/de/letsdev/maven/plugins/ios/ProvisioningProfileHelper.java

```diff
@@ -0,0 +1,179 @@
+                filepath);
+        processBuilder.directory(workDirectory);
+
+        File outputFile = File.createTempFile("outputFile.xml", "xml");
+        processBuilder.redirectOutput(outputFile);
+        CommandHelper.performCommand(processBuilder);
+
+        Document doc;
+
+        try {
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            doc = dBuilder.parse(outputFile);
+            doc.getDocumentElement().normalize();
+            String uuid = null;
+            String teamID = null;
+
+            NodeList nodeList = doc.getElementsByTagName("key");
+            for (int i = 0; i < nodeList.getLength(); i++) {
+                if (nodeList.item(i).getFirstChild().getNodeValue().equals("UUID")) {
+                    Node current = nodeList.item(i).getNextSibling();
+                    //iterate trough notes until node with value for the uuid key is current
+                    while (!current.getNodeName().equals("string")) {
+                        current = current.getNextSibling();
+                    }
+                    uuid = current.getFirstChild().getNodeValue();
+                }
+
+                if (nodeList.item(i).getFirstChild().getNodeValue().equals("TeamIdentifier")) {
+                    Node current = nodeList.item(i).getNextSibling();
+                    //iterate trough notes until array node, that has value of team id key as child is current
+                    while (!current.getNodeName().equals("array")) {
+                        current = current.getNextSibling();
+                    }
+                    current = current.getFirstChild();
+                    while (!current.getNodeName().equals("string")) {
+                        current = current.getNextSibling();
+                    }
+                    teamID = current.getFirstChild().getNodeValue();
+                }
+            }
+
+            ProvisioningProfileType type = getProvisioningProfileType(nodeList);
+
+            if (teamID != null && uuid != null) {
+                return new ProvisioningProfileData(uuid, teamID, type);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    private static String getProvisioningProfileFilePath(String provisioningProfileName) {
+
+        return provisioningProfileDirectory + "/" + provisioningProfileName + provisioningProfileFileExtension;
+    }
+
+    private static ProvisioningProfileType getProvisioningProfileType(NodeList nodeList) {
+
+        ProvisioningProfileType type = null;
+        for (int i = 0; i < nodeList.getLength(); i++) {
+
+            if (checkKey("get-task-allow", nodeList.item(i))) {
+                type = ProvisioningProfileType.TYPE_DEVELOPEMENT;
+            }
+            if (checkKey("ProvisionsAllDevices", nodeList.item(i))
```

Comment:
arbeite hier mit else if's. Dann springt er nicht noch in die anderen Zweige und du brauchst nicht das if auf type != DEVELOPMENT etc.